rfcs/theme-shape.md

```diff
@@ -0,0 +1,228 @@
+
+1. FUI library build - this is not a problem as FUI library currently does not use any global color tokens in component styles (actually there are global color tokens used in `Badge` - this is a bug, to be reported and fixed)
+2. Build of a component developed in isolation - this would cause a problem but currently, the plugin is not used here
+3. Build of an application - this is the correct place where the global color tokens are supposed to be inlined.
+
+To be safe with the FUI library build (point 1 above) and fix the potential problem when running the babel plugin on a component (point 2), we will consider the following options (**all after beta**):
+
+- Add an option to the babel plugin to fail when a global color token is used (makes point 1 safe as it will protect us from accidentally using global color tokens).
+- Modify the babel plugin to keep global color tokens as imports instead of inlining them (fixes point 2 as it moves the binding from babel-plugin in component to bundler in the application).
+- Run the babel plugin only in the application build. Neither in the FUI library nor in the component build. Process also `node_modules` (to cover the FUI library and component) during the application build.
+- Remove the babel plugin. We can also decide that the babel plugin is not worth the complexity and remove it.
+
+### <a name="merge-global-alias"></a>Merge global and alias tokens to a single object
+
+Having the tokens split to global and alias tokens is definitely a valid concept for design. But does not mean anything to engineers.
+
+**Let's merge the two to a single object in the final theme.**
+
+_(This change has no direct impact on perf, but is required for the [next point](#flatten-tokens).)_
+
+To be able to do this, we need get rid of any naming overlaps between alias and global tokens.
+The only potential overlap is in spacing tokens as we have both global and alias spacing tokens -> talk to design to understand which tokens engineers are supposed to use.
+
+Current design token spec also calls out `fontWeights` and `alignment` tokens are "not actual tokens" and should not be used -> talk to design, should we remove them?
+
+### <a name="flatten-tokens"></a>Flatten the tokens object
+
+In Typescript the theme is represented by a deep object:
+
+```js
+/* Before */
+theme.alias.colors.red.neutral;
+```
+
+This brings the three following issues:
+
+1. Engineers might struggle to find the expected token in the object.
+2. Theme merging is expensive as we need to merge deep objects.
+3. Insertion of CSS variables is more expensive as we need to flatten CSS variables in runtime (`flattenThemeToCSSVariables()`).
+
+**The proposal is to merge all the tokens to a single flat object**:
+
+```js
+/* After */
+theme.aliasColorsRedNeutral;
+```
+
+This should simplify theme merging, improve tokens discoverability. The potential drawback is we are creating a gigantic dictionary of unrelated values.
```

Comment:
Probably apreviously discarded idea, but:  Rather than declare all the values in a structure, could we have a function to pass a key and then provide all the keys in a structure for discoverability?  A JS function could allow fallback and inline just in time.  theme.color(theme.colorKeys.redNeutral)